qiskit_nature/problems/second_quantization/lattice/lattices/triangular_lattice.py

```diff
@@ -0,0 +1,272 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The triangular lattice"""
+from dataclasses import asdict
+from itertools import product
+from math import pi
+from typing import Dict, List, Optional, Tuple, Union
+
+import numpy as np
+from retworkx import PyGraph
+
+from .lattice import LatticeDrawStyle, Lattice
+from .boundary_condition import BoundaryCondition
+
+
+class TriangularLattice(Lattice):
+    """Triangular lattice."""
```

Comment:
It might be nice to have bit more of a description in these classes that describes what it is for people that might be new. For instance for me, a square lattice seem obvious with rows and columns. Not knowing what exactly this is seeing rows and columns was a bit confusing to me - maybe I am more used to triangle meshes that are used in 3d rendering. Having seen the picture in the tutorial I see how rows & cols fits in - to me it looks like the square lattice with one pair of opposite corner points being connected so the square is two triangles.

Reply:
It would certainly be better to have more explanation here (and other lattices), but we have a bit time, so let me leave it for later. Hopefully, as the first version, users will also be able to grasp the meaning from the tutorial.

Reply:
Ok, the documentation can be updated somewhat independently.